libmamba: don't log when try_lock on inexisting or non-writable path

Klaim opened the pull request.

Klaim commented:
Should help with #1702

Klaim commented on libmamba/src/core/util.cpp:
```diff
@@ -71,6 +71,16 @@ namespace mamba
+        std::error_code ec;
+        const auto status = fs::status(path, ec);
+        return !ec && status.type() == fs::file_type::not_found
+               && (status.permissions() & writable_flags) != fs::perms::none;
```
Yes, but I don't know how to check that we are the owner or not without attempting to open the file/modify the directory. So I started by only checking if at least someone can write.

Klaim replied:
I'm open to suggestion as to how to verify that, I think attempting a write is the best but also weird.

wolfv replied:
I searched a little bit and there is a function called `access(...)`. However, the manpage also says that it creates a security hole (which I am not completely understanding if it would be relevant for us): https://linux.die.net/man/2/access

Anyways, I think this PR is already an improvement, right?

wolfv replied:
Actually, we do have another `is_writable` function here:

https://github.com/mamba-org/mamba/blob/4ac84bf0f4624632e5bdbf99f842c1c621efaf12/libmamba/include/mamba/core/fsutil.hpp#L82-L101

Maybe better to use that one?

wolfv commented:
@Klaim do you want to switch to the existing `is_writable` function?

Klaim commented:
@wolfv Sorry I forgot to check that but did it right now.

Currently `mamba::path::is_writable` (the function which was there before the one I added) does not pass the tests I added for my `is_writable` implementation. I think it does the right thing by checking by creating a file, however it throws when the path matches a file, which is a different behavior than my version.

I will attempt to change the existing implementation to not throw and handle files too correctly. Hopefully no code relies on that behavior, and if it does then we could keep the new function and just add that it also checks by openging/creating the file at the given path.
